File: frontend/src/types/report.ts
export type TrafficLight = 'red' | 'yellow' | 'blue' | 'green'

export interface ReportIndicator {
  id: string
  name: string
  name_en?: string
  value: string
  unit?: string
  reference_range?: string
  light: TrafficLight
  explanation?: string
  trend?: 'rising' | 'falling' | 'stable'
}

export interface MedicalReport {
  id: string
  title: string
  hospital?: string
  report_date: string
  overall_light: TrafficLight
  summary?: string
  indicators: ReportIndicator[]
  action_plan?: ActionPlanItem[]
  status: 'processing' | 'completed' | 'failed'
  created_at: string
}

export interface ActionPlanItem {
  title: string
  description: string
  priority: TrafficLight
  category: 'checkup' | 'diet' | 'exercise' | 'medication' | 'lifestyle'
  checked?: boolean
}

export interface TrendData {
  name: string
  unit?: string
  data: { year: string; value: number }[]
}

File: backend/app/schemas/report.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime, date


class ReportUploadResponse(BaseModel):
    report_id: str
    status: str = "processing"


class IndicatorResponse(BaseModel):
    id: str
    name: str
    name_en: Optional[str] = None
    value: Optional[str] = None
    unit: Optional[str] = None
    reference_range: Optional[str] = None
    light: str
    explanation: Optional[str] = None
    trend: Optional[str] = None

    class Config:
        from_attributes = True


class ReportResponse(BaseModel):
    id: str
    title: str
    hospital: Optional[str] = None
    report_date: Optional[date] = None
    overall_light: str
    summary: Optional[str] = None
    indicators: List[IndicatorResponse] = []
    action_plan: Optional[list] = None
    status: str
    created_at: datetime

    class Config:
        from_attributes = True


class TrendDataPoint(BaseModel):
    year: str
    value: float


class TrendData(BaseModel):
    name: str
    unit: Optional[str] = None
    data: List[TrendDataPoint]
